lib/moab/storage_object_validator.rb

```diff
@@ -19,9 +20,10 @@ class StorageObjectValidator
     VERSION_DIR_BAD_FORMAT = 3
     NO_SIGNATURE_CATALOG = 4
     NO_MANIFEST_INVENTORY = 5
-    NO_XML_FILES = 6
+    NO_XML_FILES_IN_MANIFEST_DIR = 6
```

Comment:
So here's a question:  is it an error if EITHER the "metadata"  OR  the "manifests" directory has no files?   If so, you could use a NO_FILES_IN_DIR error and supply the directory as "addl" arg. 

Are you actually checking for *xml* files?  Or just files?  The KISS principle would be just check for files.